docs/src/understand/thin-edge-json.md

```diff
@@ -1,52 +1,107 @@
+You should not rely on the structure of these error messages to automate any actions as they are purely textual data
+and bound to change from time-to-time.
+
+Here is the complete list of topics reserved by `thin-edge.io` for its internal working:
+
+| Topic                                                | Description                                                            |
+|------------------------------------------------------|------------------------------------------------------------------------|
+| `tedge/`                                             | Reserved root topic of `thin-edge.io`                                  |
+| `tedge/measurements`                                 | Measurements related to the main device                                |
+| `tedge/measurements/${child-id}`                     | Measurements related to the child device named `${child-id}`           |
+| `tedge/events/${event-type}`                         | Events related to the main device                                      |
+| `tedge/events/${event-type/${child-id}`              | Events related to the child device named `${child-id}`                 |
+| `tedge/alarms/${severity}/${alarm-type}`             | Alarms related to the main device                                      |
+| `tedge/alarms/${severity}/${alarm-type}/${child-id}` | Alarms related to the child device named `${child-id}`                 |
+| `tedge/errors`                                       | Error messages emitted by `thin-edge.io` while processing measurements |
+
+### Sending Telemetry Data to the cloud
+
+The `thin-edge.io` framework allows users forward telemetry data generated and published to one of the
+`tedge/#` MQTT topics from the thin-edge device to any IoT cloud provider that it is connected to,
+with the help of a *mapper* component designed for that cloud.
+The responsibility of a mapper is to subscribe to the `tedge/#` topic to receive all incoming measurements, events and alarms
+represented in the cloud vendor neutral Thin Edge JSON format, to a format that the connected cloud understands.
+Refer to [Cloud Message Mapper Architecture](./tedge-mapper.md) for more details on the mapper component.
+
+## Measurements
+
+*Measurements* carry values from physical sensors, the domain application or monitored processes.
+For instance:
+- voltage and current of an electricity meter
+- state of the manufacturing control process
+- free disk space on the device
+
+Thin Edge JSON can be used to represent single-valued measurements, multi-valued measurements
 or a combination of both along with some auxiliary data like the timestamp at which the measurement(s) was generated.
 
-## Single-valued measurements
+### Single-valued measurements
 
 Simple single-valued measurements like temperature or pressure measurement with a single value can be expressed as follows:
 
-```json
-{
+```sh te2mqtt
+tedge mqtt pub /tedge/measurements '{
```

Comment:
```suggestion
tedge mqtt pub tedge/measurements '{
```
All the other samples in this doc also starts with `/tedge/` instead of `tedge/`. Please fix them all.